### PR TITLE
Split TooltipBubble into TooltipBubble and TooltipPopper

### PR DESCRIPTION
**Pull request by somewhatabstract**
This builds on #155 where we introduced the `TooltipArrow` component. The main purpose here is to separate the `TooltipBubble` component into a positioning component (`TooltipPopper`) and a rendering component (`TooltipBubble`).

This also adds some test examples for the bubble. The next task will be to style the bubble content.

### CODE REVIEW

File: packages/wonder-blocks-tooltip/components/tooltip-bubble.js [hunk 1]
```diff
@@ -1,77 +1,69 @@
...
+        // We only want to update the popper's arrow reference if it is
+        // actually changed. Otherwise, we end up in an endless loop of updates
+        // as every render would trigger yet another render.
+        if (popperProps && ref) {
+            const domNode = ReactDOM.findDOMNode(ref);
+            if (domNode instanceof HTMLElement && domNode !== this._lastRef) {
+                this._lastRef = domNode;
+                popperProps.ref(domNode);
+            }
+        }
```
Comment (somewhatabstract) [inline packages/wonder-blocks-tooltip/components/tooltip-bubble.js]: We changed to use `View`, so now we need to do some extra things. Also, nesting the refs means we have to be cautious not to repeatedly call the popper's ref handlers with the same element, or we'll get in a horrible state.

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 6dfbe06597785cc4c30ed033a14045f2ed854094

https://deploy-preview-157--wonder-blocks.netlify.com

File: packages/wonder-blocks-tooltip/components/tooltip-bubble.js (hunk 1 above)
Comment (kevinbarabash) [reply packages/wonder-blocks-tooltip/components/tooltip-bubble.js]: Awesome!  I'm glad there wasn't too much additional work here.

File: packages/wonder-blocks-tooltip/components/tooltip-bubble.js [hunk 2]
```diff
@@ -1,77 +1,69 @@
...
+                ]}
             >
-                {(props) => this._renderBubble(props)}
-            </Popper>
+                {children}
+                <TooltipArrow
+                    placement={placement}
+                    popperArrowProps={arrowProps}
+                />
+            </View>
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/components/tooltip-bubble.js]: So now were just rendering the bubble and the `popper` stuff is elsewhere.

File: packages/wonder-blocks-tooltip/components/tooltip-bubble.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
+This is an internal component that we use to render the stuff that appears when a tooltip shows.
+
+Note that without explicit positioning, the arrow will not be centered.
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/components/tooltip-bubble.md]: :+1:

File: packages/wonder-blocks-tooltip/components/tooltip-bubble.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+        style: {
+            left: 0,
+            top: 0,
+        },
+        ref: () => {},
+    },
+    outOfBoundaries: false,
+};
+
+<View style={{height: 50, flexDirection: "row-reverse", justifyContent: "flex-end"}}>
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/components/tooltip-bubble.md]: Using the `style` prop to override `flexDirection` won't work b/c we set `!important` on the classes applied to the underlying `div`.  In this case `flexDirection` has no effect anyways b/c there's only a single child.  Please  remove all `flexDirection`s in the examples.

File: packages/wonder-blocks-tooltip/components/tooltip-bubble.test.js [hunk 5]
```diff
@@ -7,68 +7,64 @@ import {View} from "@khanacademy/wonder-blocks-core";
...
-        const arrange = (actAssert) => {
+        const arrangeAct = (assert) => {
+            const popperProps = {
+                placement: "top",
+                style: {
+                    position: "absolute",
+                    top: 0,
+                    left: 0,
+                },
+                ref: assert,
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/components/tooltip-bubble.test.js]: Oh wow... that's a little hard to follow.  Can you add a comment for future code archeologists?

File: packages/wonder-blocks-tooltip/components/tooltip.js [hunk 6]
```diff
@@ -80,25 +81,36 @@ export default class Tooltip extends React.Component<Props, State> {
...
-                            <TooltipBubble
-                                placement={
-                                    placement || Tooltip.defaultProps.placement
-                                }
-                                anchorElement={this.state.anchorElement}
-                            >
-                                {this._renderBubbleContent()}
-                            </TooltipBubble>
-                        ) : null}
+                        {this._renderPopper(active)}
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/components/tooltip.js]: It would be good to document the full hierarchy somewhere, maybe at the top of this render function.  Is it?
```
- TooltipAnchor
-- TooltipPortalMounter
--------------------- portal divide
- TooltipPopper
-- TooltipArrow
-- TooltipBubble
--- TooltipContent
```

File: packages/wonder-blocks-tooltip/generated-snapshot.test.js [hunk 7]
```diff
@@ -192,4 +193,214 @@ describe("wonder-blocks-tooltip", () => {
...
+                ref: () => {},
+            },
+            outOfBoundaries: false,
+        };
+
+        const example = (
+            <View style={{height: 50, flexDirection: "column"}}>
+                <TooltipBubble popperProps={popperProps}>
+                    <TooltipContent>I'm on the top!</TooltipContent>
+                </TooltipBubble>
```
Comment (kevinbarabash) [inline packages/wonder-blocks-tooltip/generated-snapshot.test.js]: lol GitHub... do you even syntax highlight?

File: packages/wonder-blocks-tooltip/components/tooltip-bubble.js (hunk 2 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-tooltip/components/tooltip-bubble.js]: Yup

File: packages/wonder-blocks-tooltip/components/tooltip-bubble.md (hunk 4 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-tooltip/components/tooltip-bubble.md]: It is definitely working. Can you tell me how it won't work?

Comment (somewhatabstract) [reply packages/wonder-blocks-tooltip/components/tooltip-bubble.md]: Oh wait, I got confused. It's working in the `TooltipArrow` examples. I copied this from there. I don't need that in this case, so I should remove it.

File: packages/wonder-blocks-tooltip/components/tooltip-bubble.test.js (hunk 5 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-tooltip/components/tooltip-bubble.test.js]: Yes. This is one reason I want to do more with the popper props so that we can provide some defaults and make these tests a little nicer.

File: packages/wonder-blocks-tooltip/components/tooltip.js (hunk 6 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-tooltip/components/tooltip.js]: Yes, I agree. Hadn't done that yet since we weren't quite finished, but that's a great thing to include now.

It's:

```
- TooltipAnchor (Behavioral)
-- TooltipPortalMounter (Behavioral)
--------------------------- portal divide
- TooltipPopper (Behavioral)
-- TooltipBubble (Render)
--- TooltipContent (Render)
--- TooltipArrow (Render)
```

File: packages/wonder-blocks-tooltip/components/tooltip-bubble.test.js (hunk 5 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-tooltip/components/tooltip-bubble.test.js]: I have reworked this to hipefully clarify some things.

**Comment by somewhatabstract** [comment]
Sorry for the flog; messed up my rebasing and dropped a commit I didn't mean to.
